Fixed parsing audience (dictionary without keys)

divbyzero opened the pull request:
- Updated audience test
- Updated Extensions.swift (UIColor)
- Updated Examples (Swift & ObjC)
- Fixed type detection in EvolvRawAllocationNode
- Added check value type test (AllocationsTest)
- Fixed logging (error -> debug) in EvolvAllocations
- Added debug logging (Found key with value)

divbyzero committed:
Fixed parsing audience (dictionary without keys)
- Updated audience test
- Updated Extensions.swift (UIColor)
- Updated Examples (Swift & ObjC)
- Fixed type detection in EvolvRawAllocationNode
- Added check value type test (AllocationsTest)
- Fixed logging (error -> debug) in EvolvAllocations
- Added debug logging (Found key with value)

phylliswong reviewed (changes requested):
The button text is changing with the allocation, however I noticed that the button color assignment was #000000 yet the button color remained the yellow from the storyboard. Might also be helpful to print the allocation in the terminal somewhere to make debugging the experiment easier. Can we please add a print out of the allocation, either from the view controller or from within the SDK when the allocation is received and deserialized?

divbyzero commented:
It was the problem with a duplicate of the keys in experiments (SDK demo + store demo). "checkout.button.text". I've disabled the duplicate key and added the new ("checkout.button.title") for the store demo. In SDK I've added additional logging for debugging.

And I've fixed a problem (and added test for it) with an incorrect type that gets from an allocation. ("Int: 1" was interpreted like "bool: true").

phylliswong reviewed (approved):
thanks for making those changes
